modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingFlowComposable.kt

```diff
@@ -124,24 +126,30 @@ fun OnboardingFlowComposable(
                     },
                     onBackPressed = {
                         analyticsTracker.track(AnalyticsEvent.RECOMMENDATIONS_DISMISSED)
-                        navController.popBackStack()
+                        completeOnboarding()
                     },
                     onComplete = {
-                        // TODO analytics
-                        navController.navigate(OnboardingNavRoute.plusFeatures)
+                        navController.navigate(
+                            if (signInState?.isSignedInAsPlus == false) {
+                                OnboardingNavRoute.plusUpgrade
+                            } else {
+                                OnboardingNavRoute.welcome
+                            }
+                        )
                     }
                 )
             }
 
-            composable(OnboardingNavRoute.plusFeatures) {
-                OnboardingPlusFeaturesPage(
-                    onShown = { analyticsTracker.track(AnalyticsEvent.ONBOARDING_UPGRADE_SHOWN) },
-                    onBackPressed = {
-                        analyticsTracker.track(AnalyticsEvent.ONBOARDING_UPGRADE_DISMISSED)
-                        navController.popBackStack()
-                    },
-                    onUpgradePressed = { analyticsTracker.track(AnalyticsEvent.ONBOARDING_UPGRADE_UNLOCK_ALL_FEATUERS_TAPPED) },
-                    onNotNowPressed = { analyticsTracker.track(AnalyticsEvent.ONBOARDING_UPGRADE_NOT_NOW_TAPPED) }
+            composable(OnboardingNavRoute.plusUpgrade) {
+                OnboardingPlusUpgradeFlow(
+                    onBackPressed = { navController.popBackStack() },
+                    onNotNowPressed = { navController.navigate(OnboardingNavRoute.welcome) },
+                    onComplete = { navController.navigate(OnboardingNavRoute.welcome) },
+                )
```

Comment:
It felt to me like this navigation component shouldn't be so concerned with analytics, so I moved the analytics logic into the composable's view model. I'm planning to make this change to the other components this navigation component navigates to in a follow-up PR, so let me know if you think this isn't a good change.

Reply:
LGTM 👍